File: 백준/Silver/1193. 분수찾기/분수찾기.py
X = int(input())
sequence = 0
n = 1
donominator = 1
numerator = 1

while X > sequence:
    sequence = int(n * (n + 1) / 2)
    n += 1

if n % 2 == 0:
    denominator = n - (sequence - X) - 1
    numerator = n - denominator
else:
    numerator = n - (sequence - X) - 1
    denominator = n - numerator

print(f"{numerator}/{denominator}")

File: 백준/Silver/1193. 분수찾기/분수찾기.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);
		int X = sc.nextInt();
		sc.close();
		int n = 1;
		int sequence = 0;
		int denominator = 1;
		int numerator = 1;

		while (X > sequence) {
			sequence = n * (n + 1) / 2;
			n++;
		}

		if (n % 2 == 0) {
			denominator = n - (sequence - X) - 1;
			numerator = n - denominator;
		} else {
			numerator = n - (sequence - X) - 1;
			denominator = n - numerator;
		}
		System.out.println(numerator + "/" + denominator);
	}
}